draco/analysis/delay.py

```diff
@@ -1082,18 +1082,27 @@ def _evaluate(self, data_view, weight_view, out_cont, delays, channel_ind):
             data = data_view.local_array[lbi]
             weight = weight_view.local_array[lbi]
 
+            # Apply the window if requested
+            # Do this before applying data cut
+            # as that will change the data shape
+            if self.apply_window:
+                data *= window
+
             # Apply data cuts
             t = self._cut_data(data, weight)
             if t is None:
                 continue
             data, _, _, nzt = t
 
-            # Apply the window if requested
-            if self.apply_window:
-                data *= window
+            # Take the inverse Fourier transform
+            # Note fftw only supports complex to
+            # complex transform. For real value ringmap
+            # use numpy fft.
+            if np.iscomplexobj(data):
+                y_spec = fftw.ifft(data, axes=-1)
+            else:
+                y_spec = np.fft.ifft(data, axis=-1)
```

Comment:
```suggestion
                y_spec = np.fft.irfft(data, axis=-1)
```
Since you know the data are purely real in this case, using `irfft` should be faster, since that function will exploit the Hermitian nature of the Fourier transform.

Reply:
Although the Map is in real space (ra,dec,freq), we want to transform that to complex Fourier field (u,v,delay), and for power spectrum, we want to take complex correlation of two such fields and then take the real part of that. So, I think we want to keep the full complex quantity. This is what we do in the Wiener filter too, by providing complex_timedomain=True.

Reply:
Oh, right.  This is an inverse transform from real.  You're right, `irfft` isn't what you want.

But `ihfft` should do the trick.  It takes a purely real spectrum and outputs a (Hermitian-symmetric) complex signal.

`ifft` will work, too, but it will ignore the symmetric nature of the output (i.e. in terms of computational load, it will compute the inverse FFT as if the input were complex).

Reply:
keeping it as `ifft` for now, as this is what we are following in other parts of the code, for example the Fourier definition in Gibbs and Weiner filter code is this.